ex1/dates_times.py

```diff
@@ -26,19 +26,73 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+from datetime import date, time, datetime, timedelta
+
 '''
 Exercises for using the datetime and the calendar module
 '''
 
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
+def last_of_month(dt):
+	"""
+	Parameters
+	----------
+	dt : datetime
+
+	Returns
+	-------
+	datetime
+		last day of the month
+    	"""
+	if dt.month == 12:
+		return dt.replace(day=31)
+	return dt.replace(month=dt.month+1, day=1) - timedelta(days=1)
 
 # Define a function named feed_the_gremlin which takes a time object as an
 # argument. It should return False between midnight and 6:30AM and True
 # otherwise.
+def feed_the_gremlin(to):
+	"""
+	Parameters
+	----------
+	dt : datetime
 
+	Returns
+	-------
+	bool
+		True After 6:30AM and before midnight, else False
+    	"""
+	if( to.hour >= 0 and to.hour < 6 ):
+		return False
+	if( to.hour == 6 and to.minute < 30 ):
+		return False
+	return True
 # Define a function named how_long that takes two datetime objects dt and ref
 # where ref is the reference datetime, calculates the difference between them and
 # returns the difference as a string formatted like:
 # "01 days, 01 minutes, 01 seconds until 2000-12-31 15:59:59"
 # If ref is before dt then use 'since' instead of 'until'
+def how_long(dt, ref):
+	"""
+	Parameters
+	----------
+	dt,ref : datetime
+
+	Returns
+	-------
+	string
+		In format "01 days, 01 minutes, 01 seconds until 2000-12-31 15:59:59"
+    	"""
+	until_since = ' until '
+	time = dt - ref	# timedelta
+	if(time.days > 0):
+		until_since = ' since '
+	else:
+		time = ref - dt
+	
+	minutes = int(time.seconds/60)
+	seconds = int(time.seconds%60)
+	str_ret = "%s days, %s minutes, %s seconds" % (time.days, minutes, seconds)
```

Comment:
In this case it would be better to use `%02d` to make sure that you get the correct formatting. Using `%s` works here but only by coincidence.